csharp/src/Ice/InputStream.cs

```diff
@@ -467,16 +534,19 @@ public object ReadSerializable()
         }
 
         /// <summary>Reads a sorted dictionary from the stream.</summary>
-        /// <param name="minEntrySize">The minimum size of each entry of the dictionary, in bytes.</param>
+        /// <param name="minKeySize">The minimum size of each key of the dictionary, in bytes.</param>
+        /// <param name="minValueSize">The minimum size of each value of the dictionary, in bytes.</param>
         /// <param name="keyReader">The input stream reader used to read each key of the dictionary.</param>
         /// <param name="valueReader">The input stream reader used to read each value of the dictionary.</param>
         /// <returns>The sorted dictionary read from the stream.</returns>
         public SortedDictionary<TKey, TValue> ReadSortedDictionary<TKey, TValue>(
-            int minEntrySize,
+            int minKeySize,
+            int minValueSize,
             InputStreamReader<TKey> keyReader,
-            InputStreamReader<TValue> valueReader) where TKey : notnull
+            InputStreamReader<TValue> valueReader)
+            where TKey : notnull
         {
-            int sz = ReadAndCheckSeqSize(minEntrySize);
+            int sz = ReadAndCheckSeqSize(minKeySize + minValueSize);
```

Comment:
Couldn't we share the same impl for dictionary / sorted dictioannary?

Reply:
Yes, fixed.